src/deployment.ts

```diff
@@ -227,6 +228,18 @@ export class Deployment extends workload.Workload {
     };
   }
 
+  /**
+   * @see IScalable.toScalingTarget()
+   */
+  public toScalingTarget(): ScalingTarget {
+    return {
+      kind: this.apiObject.kind,
+      apiVersion: this.apiObject.apiVersion,
+      name: this.name,
+      containers: this.containers,
+      replicas: this.replicas ?? 0,
```

Comment:
Why use `0` as a default? Seems dangerous, won't `undefined` work just as well?

Reply:
Sure we can accept undefined here. I only did this to keep the interface type tight. Although out of curiosity can you explain why setting `0` here would be dangerous?

Reply:
Updated the `ScalingTarget` to accept undefined